modules/admin-ui/src/main/webapp/scripts/modules/events/controllers/eventController.js

```diff
@@ -882,15 +882,60 @@ angular.module('adminNg.controllers')
         checkForActiveTransactions();
 
         $scope.workflowAction = function (wfId, action) {
-        	EventWorkflowActionResource.save({id: $scope.resourceId, wfId: wfId, action: action}, function () {
-                Notifications.add('success', 'EVENTS_PROCESSING_ACTION_' + action);
-                $scope.modal_close();
+            if ($scope.workflowActionInProgress) return;
+            $scope.workflowActionInProgress = true;
+            EventWorkflowActionResource.save({id: $scope.resourceId, wfId: wfId, action: action}, function () {
+                Notifications.add('success', 'EVENTS_PROCESSING_ACTION_' + action, NOTIFICATION_CONTEXT);
+
+                // We update our client-side model in case of success, so we don't have to send a new request
+                if ($scope.workflows.entries) {
+                    var wf = $scope.workflows.entries.find(function (wf) {
+                        return wf.id === wfId;
+                    });
+                    var actionUpper = action.toUpperCase();
+                    switch (actionUpper) {
+                        case 'NONE':
+                        case 'RETRY':
+                            wf.status = 'RUNNING';
+                            break;
```

Comment:
What I noticed is that the event details dialog will say 'RUNNING' even if the workflow has not been resumed yet by the WorkflowService (you can see its state in the events list in the background). Also, if the dialog remains open and the workflow fails, it will remain saying 'RUNNING". So I think we either make sure the state shown in the event details dialog is accurate or we need to close the dialog.

Reply:
Ok, I think it is now acceptable to close the dialog since we don't have the pause button anymore and updating it correctly would mean a higher effort. I changed it back to closing the modal instead trying to update the content.